src/main/java/org/spongepowered/common/data/manipulator/mutable/item/SpongeDurabilityData.java

```diff
@@ -0,0 +1,116 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.mutable.item;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ComparisonChain;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableDurabilityData;
+import org.spongepowered.api.data.manipulator.mutable.item.DurabilityData;
+import org.spongepowered.api.data.value.mutable.MutableBoundedValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.immutable.item.ImmutableSpongeDurabilityData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.value.SpongeValueBuilder;
+
+public class SpongeDurabilityData extends AbstractData<DurabilityData, ImmutableDurabilityData> implements DurabilityData {
+
+    int durability;
+    boolean unbreakable;
+
+    public SpongeDurabilityData() {
+        this(60, false);
+    }
+
+    public SpongeDurabilityData(int durability, boolean unbreakable) {
+        super(DurabilityData.class);
+        checkArgument(durability >= 0);
+        this.durability = durability;
+        this.unbreakable = unbreakable;
+        this.registerGettersAndSetters();
+    }
+
+    @Override
+    protected void registerGettersAndSetters() {
+        registerFieldGetter(Keys.ITEM_DURABILITY, () -> this.durability);
+        registerFieldSetter(Keys.ITEM_DURABILITY, this::setDurability);
+        registerKeyValue(Keys.ITEM_DURABILITY, this::durability);
+
+        registerFieldGetter(Keys.UNBREAKABLE, () -> this.unbreakable);
+        registerFieldSetter(Keys.UNBREAKABLE, value -> this.unbreakable = value);
+        registerKeyValue(Keys.UNBREAKABLE, this::unbreakable);
+    }
+
+    public void setDurability(int durability) {
+        if (durability >= 0) {
```

Comment:
replace with `checkArgument(durability >= 0, "Durability cannot be less than zero!");`